src/x25519.rs

```diff
@@ -63,6 +77,22 @@ impl EphemeralSecret {
         EphemeralSecret(clamp_scalar(bytes))
     }
 
+    /// View this secret as an array of bytes.
+    pub fn as_bytes<'a>(&'a self) -> &'a [u8; 32] {
+        self.0.as_bytes()
+    }
+
+    /// Converts this secret to its raw byte array representation.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0.to_bytes()
+    }
+}
+
+impl From<[u8; 32]> for EphemeralSecret {
+    /// Given a byte array, construct an `EphemeralSecret`.
+    fn from(bytes: [u8; 32]) -> EphemeralSecret {
+        EphemeralSecret(clamp_scalar(bytes))
+    }
```

Comment:
If you can extract the secret bytes from an `EphemeralSecret`, and then feed them into a new `EphemeralSecret` object, it's not guaranteed to be ephemeral any more. So I'm not sure this is a good change.